ecpy/measure/editors/execution_editor/editor.enaml

```diff
@@ -289,54 +279,39 @@ enamldef ComplexTaskExecutionEditor(GroupBox): main:
     #: Reference to the model driving the whole execution property edition.
     attr _model
 
+    #: List of task that should be displayed by the editor. Keep cached
+    #: to determine the modifications
+    attr _tasks = OrderedDict()
+
     func refresh():
         """Function making sure that the editor is displayed correctly.
 
         """
-        self.show()
-        for ed in tag.objects:
-            ed.set_parent(main)
+        # First update the children before showing the widget
+        for ed in _tasks.values():
+            ed.set_parent(c_tag)
             ed.refresh()
+        c_tag.request_relayout()
+        self.show()
 
     func populate_tagged_children(change=None):
         """Refresh the editors for the included tagged children.
 
         """
-        if isinstance(change, dict) and 'oldvalue' in change:
-            if isinstance(change['oldvalue'], Iterable):
-                removed = set(change['oldvalue']) - set(change['value'])
-                for t in removed:
-                    root.discard_view(t)
-            else:
-                root.discard_view(change['oldvalue'])
-        elif change is not None:
-            if change.collapsed:
-                for c in change.collapsed:
-                    for _, t in c.removed:
-                        root.discard_view(t)
-            else:
-                for _, t in change.removed:
-                    root.discard_view(t)
+        tasks = task.gather_children()
+        added = set(tasks) - set(self._tasks)
+        removed = set(self._tasks) - set(tasks)
+        self._tasks = OrderedDict([(t, root.view_for(t)) for t in tasks])
 
-        views = []
-        for child in task.gather_children():
-            v = root.view_for(child)
-            v.visible = main.show_children
-            views.append(v)
+        for t in removed:
+            root.discard_view(t)
 
-        tag.objects = views
+        for t in added:
+            _tasks[t].set_parent(c_tag)
 
-    title = task.name if task else ''
+        c_tag.request_relayout()
```

Comment:
It is a method of a parent class (https://github.com/MatthieuDartiailh/enaml/blob/master/enaml/widgets/constraints_widget.py#L73) to request a new calculation of the layout. This is needed because reordering tasks does not change the children of the container but should change the display and there is no automatic notification triggering it.